spec/integration/userposts_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+RSpec.feature 'User post index page', type: :feature do
+  let(:user) { FactoryBot.create(:user) }
+  before do
+    # Create some posts, comments, and likes for the user
+    FactoryBot.create_list(:post, 5, author: user)
+    visit user_posts_path(user)
+  end
+  scenario 'Check if user profile picture is visible' do
+    expect(page).to have_css('img')
+  end
+  scenario 'Check if users username is visible' do
+    expect(page).to have_content(user.name)
+  end
+  scenario 'Check the number of posts the user has written.' do
+    expect(page).to have_content("Number Of posts #{user.posts_counter}")
+  end
+  scenario 'Check if post titles and bodies are visible' do
+    user.posts.each do |post|
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.text.truncate(50))
+      expect(page).to have_content(post.comments.first.body) if post.comments.any?
+      expect(page).to have_content("Comments #{post.comments.count}")
+      expect(page).to have_content("Likes #{post.likes.count}") # Truncate post body for visibility
+    end
+  end
+  scenario 'Check if the pagination section is visible' do
+    expect(page).to have_css('.pagination')
+  end
```

Comment:
Nice job so far👍🏼. As a requirement, all tests must be passed. kindly ensure all tests are passing before proceeding to submit😉.

<img width="822" alt="image" src="https://github.com/erickma1/blog_app/assets/64643995/92745d42-ed47-4f5e-aa7f-92ddde8c7e63">